runtime/runtime/src/lib.rs

```diff
@@ -2020,7 +2020,10 @@ impl Runtime {
             delayed_receipts.apply_congestion_changes(congestion_info)?;
             let all_shards = apply_state.congestion_info.all_shards();
 
-            let congestion_seed = apply_state.block_height.wrapping_add(apply_state.shard_id);
+            // TODO(wacban) Using non-contiguous shard id here breaks some
+            // assumptions. The shard index should be used here instead.
+            let congestion_seed =
```

Comment:
What is the worst that can happen? The allowed shard is not chosen fairly?

Reply:
It could lead to some shards never be the allowed shard. I'm not sure as I'm avoiding digging deeper into those non-trivial cases as this task is massive as it stands. For now a todo should do :)